tmt/steps/provision/testcloud.py

```diff
@@ -972,6 +983,22 @@ def start(self) -> None:
 
         self._apply_hw_arch(self._domain, self.is_kvm, self.is_legacy_os)
 
+        # Do we have virtiofs to handle?
+        if self.virtiofs:
+            if (f"qemu:///{self.connection}" == "qemu:///session" and
+                    libvirt.getVersion() < 10000000):
+                raise tmt.utils.ProvisionError("Only system connection is supported"
+                                               " for virtiofs on libvirt < 10.0.")
+
+            virtiofs_split = self.virtiofs.split(":")
+            if len(virtiofs_split) != 2:
+                raise tmt.utils.ProvisionError("virtiofs specified in a bad format, use "
+                                               "<host path>:<guest path>")
+
+            self._domain.virtiofs_configuration = [
```

Comment:
Right, this processing should go into a custom normalization method, and plugin should work with lists and tuples or dataclasses. Validate & normalize input from fmf and CLI into a single internal representation.